pkg/nsx/services/common/store.go

```diff
@@ -62,6 +62,33 @@ func DecrementPageSize(pageSize *int64) {
 	}
 }
 
+func (resourceStore *ResourceStore) ListIndexFuncValues(key string) sets.String {
+	values := sets.NewString()
+	entities := resourceStore.Indexer.ListIndexFuncValues(key)
+	for _, entity := range entities {
+		values.Insert(entity)
+	}
+	return values
+}
+
+func (resourceStore *ResourceStore) GetByKey(key string) interface{} {
```

Comment:
As we discussed, we can move GetByKey and GetByIndex to Store interface since they are also generic interfaces needed by store. 
In addition, please add some comments to explain a bit of these function purpose.

Reply:
As discussed.